packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/V1_ProtocolToMetaModelPropertyMappingVisitor.ts

```diff
@@ -122,7 +136,33 @@ export class V1_ProtocolToMetaModelPropertyMappingVisitor
       'Model-to-model property mapping transform lambda is missing',
     );
     // NOTE: mapping for derived property is not supported
-    const property = this.context.resolveProperty(protocol.property);
+    let property: PropertyReference;
+    let localMapping: LocalMappingPropertyInfo | undefined;
+    if (protocol.localMappingProperty) {
```

Comment:
can you remind me why we only need to do this for M2M and not flat-data/relational?

Reply:
It is not supported via a FlatData Property Mapping. For relational, there is special processing leveraging the mapping class. I would rather wait until that is available in legend engine for proper testing and the proper roundtrip test to be added.